keel-igor/src/main/kotlin/com/netflix/spinnaker/igor/BuildService.kt

```diff
@@ -0,0 +1,25 @@
+package com.netflix.spinnaker.igor
+import com.netflix.spinnaker.keel.api.artifacts.Build
+import retrofit2.http.GET
+import retrofit2.http.Query
+
+interface BuildService {
+
+  /**
+   * Retrieves build information by commit id and build number from a ci service.
+   *
+   * @param projectKey The "project" within the SCM system where the repository exists, which can be a user's personal
+   *        area (e.g. "SPKR", "~<username>")
+   * @param repoSlug The repository name (e.g. "myapp")
+   * @param commitId The commit id.
+   * @param buildNumber the build number .
+   */
+  // TODO: implement this call in igor
+  @GET("/ci/builds")
```

Comment:
I think you'll need to put in retries and handle error from this call. For example, if the build is not found the convention in spinnaker services is to throw a 404. you'll need to catch that and do the appropriate thing

Reply:
I thought about it, but I _think_ for any kind of exception, I will want to retry as the data might be there later on from some reason.. let me know if it makes sense.